src/Trakx.Shrimpy.ApiClient/AddShrimpyClientExtensions.cs

```diff
@@ -22,7 +26,67 @@ public static IServiceCollection AddShrimpyClient(this IServiceCollection servic
         serviceCollection.AddSingleton<IShrimpyCredentialsProvider, ApiKeyCredentialsProvider>();
         serviceCollection.AddSingleton(apiConfiguration);
         serviceCollection.AddSingleton<ClientConfigurator>();
-        AddClients(serviceCollection);
+
+        serviceCollection.ConfigureHttpClient<IMarketDataClient, MarketDataClient>();
+        serviceCollection.ConfigureHttpClient<IAccountsClient, AccountsClient>();
```

Comment:
These lines configure the `HttpClient` as well as the retry policy via `Polly`.

We have been doing this with a T4 code generator template.
For `Kaiko.ApiClient`, I removed the T4 template and wrote some helper generic methods.
I have now copied them into this PR.

If the team find these methods easier to manage than the T4 templates, I will submit a PR to `Common.ApiClient` with them.